### DISCUSSION TITLE
Using Personal Access Tokens

### DISCUSSION THREAD

**Discussion by IROdEvO** [community/community#23402] [New to GitHub]
<p>Today, there was a scheduled brownout. Within that period it was not possible to pull, push using git username and password. My question is, how to use a Personal Access Token to clone, pull, push from GitHub? What are the terminal commands?</p>
<p>I’m using Linux and using the terminal to push to and pull from my GitHub repositories.</p>
<p>Typing the Personal Access Token instead of the password did not work either within the brownout period.</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/irodevo">@IROdEvO</a>, welcome to the community.<br>
It definitely works, it you provide the details of what typed/commands entered and what responses received we should be able to solve any problems.  Please do NOT provide any credentials (PAT or password).<br>
Or lool at one of the may other resolutions seen for other useers with similar questions such as</p><aside class="quote quote-modified" data-post="1" data-topic="163304">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-to-start-using-a-personal-access-token-to-authenticate-git-on-windows/163304">How to start using a personal access token to authenticate git on windows</a> <a class="badge-wrapper  bullet" href="/c/github-help/how-to-use-git-and-github/21"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #ffd1ac;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Ask and answer questions regarding Git and GitHub.">How to use Git and GitHub</span></a>
  </div>
  <blockquote>
    I have  created a personal access token to replace the use of password using git over hhtps. From the help docs I’ve read on the subject, I can’t find straight forward information about how to start using the PAT and replace my current form of authentcaition which is a password. When I perform git operations, I am not being prompted for a password so I guess the windows credential manager is managing the password authentication for me. So what do I need to do to actually make the switch? And wha…
  </blockquote>
</aside>

**Comment by jsoref** [comment, marked as answer]
<p>If you’re on Linux, I don’t understand why you aren’t just setting up an ssh key.</p>
<p><a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh</a></p>

**Comment by IROdEvO** [comment]
<p>I set up an SSH key. It seems to be the most convenient way. Thank you very much :blush:</p>
